File: The-Full-Counting-Sort.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{

    /*
     * Complete the 'countSort' function below.
     *
     * The function accepts 2D_STRING_ARRAY arr as parameter.
     */

    public static void countSort(List<List<string>> arr)
    {
        int n = arr.Count;
        var buckets = new List<string>[100];
        
        for (int i = 0; i < n; i++)
        {
            int key = int.Parse(arr[i][0]);
            string value = (i < n / 2) ? "-" : arr[i][1];
            
            if (buckets[key] == null)
                buckets[key] = new List<string>();
            
            buckets[key].Add(value);
        }
        
        var result = new StringBuilder();
        for (int i = 0; i < 100; i++)
        {
            if (buckets[i] != null)
            {
                foreach (var str in buckets[i])
                {
                    result.Append(str).Append(" ");
                }
            }
        }
        
        Console.Write(result.ToString().TrimEnd());        
    }


    public static void countSort2(List<List<string>> arr)
    {
        int n = arr.Count;
        var buckets = new Dictionary<int, List<string>>();
        
        for (int i = 0; i < n; i++)
        {
            int key = int.Parse(arr[i][0]);
            string value = (i < n / 2) ? "-" : arr[i][1];
            
            if (!buckets.ContainsKey(key))
                buckets[key] = new List<string>();
            
            buckets[key].Add(value);
        }
        
        var result = new StringBuilder();
        for (int i = 0; i < 100; i++)
        {
            if (buckets.ContainsKey(i))
            {
                foreach (var str in buckets[i])
                {
                    result.Append(str).Append(" ");
                }
            }
        }
        
        Console.Write(result.ToString().TrimEnd());
    }
}


class Solution
{
    public static void Main(string[] args)
    {
        int n = Convert.ToInt32(Console.ReadLine().Trim());

        List<List<string>> arr = new List<List<string>>();

        for (int i = 0; i < n; i++)
        {
            arr.Add(Console.ReadLine().TrimEnd().Split(' ').ToList());
        }

        Result.countSort(arr);
    }
}

File: The-Full-Counting-Sort.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'countSort' function below.
     *
     * The function accepts 2D_STRING_ARRAY arr as parameter.
     */

    public static void countSort(List<List<String>> arr) {
        int n = arr.size();
        List<List<String>> buckets = new ArrayList<>(100);
        for (int i = 0; i < 100; i++) {
            buckets.add(new ArrayList<>());
        }

        for (int i = 0; i < n; i++) {
            int key = Integer.parseInt(arr.get(i).get(0));
            String value = (i < n / 2) ? "-" : arr.get(i).get(1);
            buckets.get(key).add(value);
        }

        StringBuilder result = new StringBuilder();
        for (int i = 0; i < 100; i++) {
            if (buckets.get(i) != null) {
                for (String str : buckets.get(i)) {
                    result.append(str).append(" ");
                }
            }
        }
        
        System.out.print(result.toString().trim());
    }


    public static void countSort2(List<List<String>> arr) {
        int n = arr.size();
        Map<Integer, List<String>> buckets = new HashMap<>();
        
        for (int i = 0; i < n; i++) {
            int key = Integer.parseInt(arr.get(i).get(0));
            String value = (i < n / 2) ? "-" : arr.get(i).get(1);
            
            if (!buckets.containsKey(key)) {
                buckets.put(key, new ArrayList<>());
            }
            
            buckets.get(key).add(value);
        }
        
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < 100; i++) {
            if (buckets.containsKey(i)) {
                for (String str : buckets.get(i)) {
                    result.append(str).append(" ");
                }
            }
        }
        
        System.out.print(result.toString().trim());
    }
}


public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<List<String>> arr = new ArrayList<>();

        IntStream.range(0, n).forEach(i -> {
            try {
                arr.add(
                    Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
                        .collect(toList())
                );
            } catch (IOException ex) {
                throw new RuntimeException(ex);
            }
        });

        Result.countSort(arr);

        bufferedReader.close();
    }
}
